codec/src/main/java/io/netty/handler/codec/compression/Zstd.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.compression;
+
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+public final class Zstd {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(Zstd.class);
+    private static final Throwable cause;
+
+    static {
+        Throwable t = null;
+
+        try {
+            Class.forName("com.github.luben.zstd.Zstd", false,
+                PlatformDependent.getClassLoader(Zstd.class));
+        } catch (ClassNotFoundException e) {
+            t = e;
+            logger.debug(
+                "zstd-jni not in the classpath; Zstd support will be unavailable.");
+        } catch (ExceptionInInitializerError e) {
+            t = e.getCause();
```

Comment:
I think we should not unwrap this one...  Which all omeans we could just remove the catch block as it will be handled in `catch Throwable`

Reply:
My only concern here was to make sure we have an `UnsatisfiedLinkError` instead of an `ExceptionInInitializerError` as cause but clearly this made things a bit redundant.